types/sinon/ts3.1/index.d.ts

```diff
@@ -351,11 +353,23 @@ declare namespace Sinon {
          * The original method can be restored by calling object.method.restore().
          * The returned spy is the function object which replaced the original method. spy === object.method.
          */
-        <T, K extends keyof T>(obj: T, method: K, types?: string[]): T[K] extends (
+        <T, K extends keyof T>(obj: T, method: K): T[K] extends (
             ...args: infer TArgs
         ) => infer TReturnValue
             ? SinonSpy<TArgs, TReturnValue>
             : SinonSpy;
+
+        <T, K extends keyof T>(obj: T, method: K, types: PropertyAccessorType[]): {
+            [P in PropertyAccessorType]?: T[K] extends (
+                ...args: infer TArgs
+            ) => infer TReturnValue
+                    ? SinonSpy<TArgs, TReturnValue>
+                    : SinonSpy;
+        } & {
```

Comment:
if possible the object should be an extension of a `PropertyDescriptor` in some way. we don't want to be maintaining our own copy of what a descriptor looks like